assets/blocks/editor-components/confirm-dialog/confirm-dialog.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import { Modal, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
+import { ENTER } from '@wordpress/keycodes';
+
+/**
+ * Controlled Component that shows a modal containing a confirm dialog. Inspired by Gutenberg's experimental
+ * Confirm Dialog.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/components/confirm-dialog/
+ * @param {Object}   props           Component props.
+ * @param {boolean}  props.isOpen    Determines if the confirm dialog is open or not
+ * @param {string}   props.title     Title for the confirm dialog. Default is window.location.host value.
+ * @param {string}   props.children  Content for the confirm dialog, can be any React component.
+ * @param {Function} props.onConfirm Callback called when the user click on "OK" or press Enter with the modal open.
+ * @param {Function} props.onCancel  Callback called when the user click on "Cancel" or press ESC with the modal open.
+ */
+const ConfirmDialog = ( {
+	isOpen = false,
+	title = window.location.host,
+	children,
+	onConfirm,
+	onCancel,
+} ) => {
+	useConfirmOnEnter( isOpen, onConfirm );
+	if ( ! isOpen ) {
+		return null;
+	}
+	return (
+		<Modal
+			title={ title }
+			onRequestClose={ onCancel }
+			shouldCloseOnClickOutside={ false }
+			className="sensei-confirm-dialog"
+		>
+			<div className="sensei-confirm-dialog__message">{ children }</div>
+			<div className="sensei-confirm-dialog__button-container">
+				<Button variant="tertiary" onClick={ onCancel }>
+					{ __( 'Cancel', 'sensei-lms' ) }
+				</Button>
+				<Button variant="primary" onClick={ onConfirm }>
+					{ __( 'OK', 'sensei-lms' ) }
+				</Button>
+			</div>
+		</Modal>
+	);
+};
+
+/**
+ * Calls onConfirm when registerListener is true and the user press ENTER.
+ *
+ * @param {boolean}  registerListener If the listener should be set up or not.
```

Comment:
Renamed to `shouldRegisterListener` in [0a0741b](https://github.com/Automattic/sensei/pull/5361/commits/0a0741bec90c3419b20633f3ceca51cfec44933f).

I thought about using `isOpen`  here - because that's the value that's actually passed to the method, but...IDK, I think a more explicit name should be good here.